op-bindings/bindgen/generator_remote.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/etherscan"
+)
+
+type bindGenGeneratorRemote struct {
+	bindGenGeneratorBase
+	contractDataClients map[string]contractDataClient
+	tempArtifactsDir    string
+}
+
+type contractDataClient interface {
+	FetchAbi(address string) (string, error)
+	FetchDeployedBytecode(address string) (string, error)
+	FetchDeploymentTxHash(address string) (string, error)
+	FetchDeploymentTx(txHash string) (etherscan.TxInfo, error)
+}
+
+type remoteContract struct {
+	Name            string            `json:"name"`
+	Verified        bool              `json:"verified"`
+	Deployments     map[string]string `json:"deployments"`
+	DeploymentSalt  string            `json:"deploymentSalt"`
+	DeployerAddress string            `json:"deployerAddress"`
```

Comment:
`DeployerAddress` could just be `Deployer` and then make the value be a `common.Address`. No need to encode the type into the name of the field